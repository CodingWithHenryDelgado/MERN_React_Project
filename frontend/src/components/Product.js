import React from 'react'
import { Link } from 'react-router-dom'
import { Card } from 'react-bootstrap'

const Product = ({ product }) => {
    return (
        <Card className='my-3 p-3 rounded'>
            <Link to={`/product/${product._id}`} style={{ width: '100%' }}>
                <Card.Body style={{ width: '100%', height: 225 }} className="d-flex align-items-center justify-content-center">
                    <Card.Img className="mx-3 py-4" src={product.image} variant='top' style={{ maxWidth: '60%', height: 'auto' }} />
                </Card.Body>
            </Link>

            <Card.Body>
                <Link to={`/product/${product._id}`}>
                    <Card.Title as='div'>
                        <strong>{product.name}</strong>
                    </Card.Title>
                </Link>
            </Card.Body>
        </Card>
    )
}

export default Product