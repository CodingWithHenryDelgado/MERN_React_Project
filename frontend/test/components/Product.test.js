import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { Card } from 'react-bootstrap';
import Product from '../../src/components/Product';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

//Wojtekmaj module bridges Enzyme with React 17.
Enzyme.configure({ adapter: new Adapter() });

describe('Product component', () => {
    const product = {
        _id: '1',
        name: 'Test Product',
        image: 'testimage.jpg',
    };

    it('should render the component correctly', () => {
        const wrapper = shallow(<Product product={product} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain a Link with the correct URL', () => {
        const wrapper = shallow(<Product product={product} />);
        const link = wrapper.find(Link).first();
        expect(link.prop('to')).toEqual(`/product/${product._id}`);
    });

    it('should contain a Card with the correct image source and title', () => {
        const wrapper = shallow(<Product product={product} />);
        const cardImage = wrapper.find(Card.Img);
        const cardTitle = wrapper.find(Card.Title);
        expect(cardImage.prop('src')).toEqual(product.image);
        expect(cardTitle.text()).toEqual(product.name);
    });
});
