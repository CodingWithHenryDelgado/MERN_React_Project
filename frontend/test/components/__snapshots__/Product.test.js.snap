// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render the component correctly 1`] = `ShallowWrapper {}`;
